packages/contentful/src/utils.ts

```diff
@@ -1,5 +1,53 @@
-import { Document } from '@contentful/rich-text-types';
+import { Document, Node } from '@contentful/rich-text-types';
 import { documentToHtmlString } from '@contentful/rich-text-html-renderer';
 
-export const parseRichText = (document: Document) =>
-  documentToHtmlString(document);
+export type RichTextFromQuery = {
+  json: Document;
+  links: {
+    assets: {
+      block?: {
+        sys: {
+          id: string;
+        };
+        url: string;
+        description?: string;
+      }[];
+    };
+  };
+};
+
+type AssetById = {
+  [assetId: string]: {
+    alt?: string;
+    url: string;
+  };
+};
+
+export const parseRichText = (rtf: RichTextFromQuery) => {
+  const assetById: AssetById | undefined = rtf.links.assets.block?.reduce(
+    (assetInfoById: AssetById, asset) => ({
+      ...assetInfoById,
+      [asset.sys.id]: {
+        alt: asset.description as string | undefined,
+        url: asset.url,
+      },
+    }),
+    {},
+  );
+
+  const options = {
+    renderNode: {
+      'embedded-asset-block': (node: Node) => {
+        const assetId = node.data.target.sys.id as string;
+        if (assetById && assetId && assetById[assetId]) {
```

Comment:
what if its a different asset like pdf?

Reply:
in contentful it seems it's the same :) so I just needed to update this `embedded-asset-block` render node